### ISSUE TITLE
register_post_expansion error: "type annotations required"

### ISSUE THREAD

**Issue by keeperofdakeys** [dtolnay/post-expansion#1]
I'm getting a strange error when trying to use this with the latest nightly, I've had a look at the pretty=expanded output, but I can't quite see what it's referring to. I've uploaded the code to my [asn1-rs repo](https://github.com/keeperofdakeys/asn1-rs) (it's in the asn1-cereal-derive directory).

```
error[E0284]: type annotations required: cannot resolve `<_ as std::str::FromStr>::Err == _`
  --> /home/blueteeth/Applications/projects/asn1-rs/asn1-cereal-derive/src/lib.rs:22:1
   |
22 | register_post_expansion!(PostExpansion_asn1info);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate
```

**Comment by dtolnay** [comment]
It would help if I release the code that I test, rather than testing then "cleaning up" the code then releasing. Fixed in 8a7b140799689cf17ba84715fa0517d9bf32d4d4 :sweat_smile: and I released 0.0.2.

It passes Serde's test suite in https://github.com/serde-rs/serde/pull/592 (on my computer; I have to head to work before waiting for the Travis build).

**Comment by keeperofdakeys** [comment]
As far as I can see, this is working for me now. Thanks.
